File: native/iosApp/iosApp/Components/StatusBadge.swift
import SwiftUI

struct StatusBadge: View {
    let status: String
    let label: String

    private var bgColor: Color {
        switch status {
        case "active", "signed": return Color.appSuccess.opacity(0.15)
        case "pending", "draft": return Color(hex: "F59E0B").opacity(0.15)
        case "inactive", "rejected", "revoked": return Color.appDestructive.opacity(0.15)
        default: return Color.appMuted.opacity(0.15)
        }
    }

    private var textColor: Color {
        switch status {
        case "active", "signed": return .appSuccess
        case "pending", "draft": return Color(hex: "F59E0B")
        case "inactive", "rejected", "revoked": return .appDestructive
        default: return .appMuted
        }
    }

    var body: some View {
        Text(label)
            .font(.appCaption)
            .fontWeight(.medium)
            .foregroundColor(textColor)
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(bgColor)
            .cornerRadius(CGFloat(AppTheme.Radius.sm))
            .accessibilityIdentifier("status-badge-\(status)")
    }
}

File: native/androidApp/src/main/kotlin/com/imirae/incheon/ui/components/StatusBadge.kt
package com.imirae.incheon.ui.components

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.imirae.incheon.design.DesignTokens

@Composable
fun StatusBadge(status: String, label: String, modifier: Modifier = Modifier) {
    val (bgColor, textColor) = when (status) {
        "active", "signed" -> Color(DesignTokens.Colors.success).copy(alpha = 0.15f) to Color(DesignTokens.Colors.success)
        "pending", "draft" -> Color(DesignTokens.Colors.warning).copy(alpha = 0.15f) to Color(DesignTokens.Colors.warning)
        "inactive", "rejected", "revoked" -> Color(DesignTokens.Colors.destructive).copy(alpha = 0.15f) to Color(DesignTokens.Colors.destructive)
        else -> Color(DesignTokens.Colors.muted).copy(alpha = 0.15f) to Color(DesignTokens.Colors.mutedForeground)
    }
    Surface(
        color = bgColor,
        shape = RoundedCornerShape(DesignTokens.Radius.sm),
        modifier = modifier.testTag("status-badge-$status")
    ) {
        Text(
            text = label,
            color = textColor,
            style = MaterialTheme.typography.bodySmall.copy(fontWeight = FontWeight.Medium),
            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp)
        )
    }
}
